src/app/products/products.service.ts

```diff
@@ -98,18 +98,31 @@ export class ProductsService {
   }
 
   /**
-   * @returns {Observable<any>} Charges data
+   * @returns {Observable<any>} Charges data.
    */
   getCharges(): Observable<any> {
     return this.http.get('/charges');
   }
 
+  getChargesTemplate(): Observable<any> {
+    return this.http.get('/charges/template');
+  }
+
   /**
    * @param {string} chargeId Charge ID of charge.
    * @returns {Observable<any>} Charge.
    */
-  getCharge(chargeId: string): Observable<any> {
-    return this.http.get(`/charges/${chargeId}`);
+  getCharge(selectedCharge: string, template: boolean = false): Observable<any> {
```

Comment:
If there is a need to retrieve data from any API, you can declare it in another function without changing to the previous one, in case it may also hamper any other module. If any other reason, please specify.

Reply:
That template parameter was needed to get the available options for the drop downs. So the relevant resolver function passes the boolean true for the template argument. But in view the default value will apply. Do I have to create separate functions for that?

Reply:
Okay. Let it be.